include/camp/resource.hpp

```diff
@@ -155,6 +155,24 @@ namespace resources
     };
 #endif
 
+  template<typename Res>
+  struct EventProxy {
+    EventProxy(EventProxy &&) = default;
+    EventProxy(EventProxy const &) = delete;
+    EventProxy &operator=(EventProxy &&) = default;
+    EventProxy &operator=(EventProxy const &) = delete;
+
+    EventProxy(Res* r) :
+      resource_{r}
+    {}
+
+    operator Event() const {
```

Comment:
Also please add an explicit `get` here.  I think having the conversion is fine, but I always prefer having an explicit option there as well.